lib/activerecord_sqlserver_adapter/jdbc/dataset.rb

```diff
@@ -0,0 +1,225 @@
+module ActiveRecordSqlServerAdapter
+  module Jdbc
+    class Dataset
+      include Enumerable
+
+      OPTS = {}.freeze
+
+      def synchronize(&block)
+        @cache_mutex.synchronize(&block)
+      end
```

Comment:
The jdbc connection should be thread save, but our ruby jdbc wrapper is not.

One example is the implementation of Database#execute which does executes two statements in one after another for inserts on the same connection. If this connection would be used in between these statements the result of the second statement would be probably wrong.

Maybe this is not necessary because AR does pooling anyways, but I'm not sure.